core/domain/skill_domain.py

```diff
@@ -247,6 +322,71 @@ def __init__(
         self.last_updated = last_updated
         self.version = version
 
+    def validate(self):
+        """Validates various properties of the Skill object.
+
+        Raises:
+            ValidationError: One or more attributes of skill are invalid.
+        """
+        if not isinstance(self.description, basestring):
+            raise utils.ValidationError(
+                'Expected description to be a string, received %s'
+                % self.description)
+
+        if not isinstance(self.misconceptions_schema_version, int):
+            raise utils.ValidationError(
+                'Expected misconceptions schema version to be an integer, ' +
+                'received %s' % self.misconceptions_schema_version)
+        if (
+                self.misconceptions_schema_version !=
+                feconf.CURRENT_MISCONCEPTIONS_SCHEMA_VERSION):
+            raise utils.ValidationError(
+                'Expected misconceptions schema version to be %s, received %s'
+                % (
+                    feconf.CURRENT_MISCONCEPTIONS_SCHEMA_VERSION,
+                    self.misconceptions_schema_version)
+            )
+
+        if not isinstance(self.skill_contents_schema_version, int):
+            raise utils.ValidationError(
+                'Expected skill contents schema version to be an integer, ' +
+                'received %s' % self.skill_contents_schema_version)
+        if (
+                self.skill_contents_schema_version !=
+                feconf.CURRENT_SKILL_CONTENTS_SCHEMA_VERSION):
+            raise utils.ValidationError(
+                'Expected skill contents schema version to be %s, received %s'
+                % (
+                    feconf.CURRENT_SKILL_CONTENTS_SCHEMA_VERSION,
+                    self.skill_contents_schema_version)
+            )
+
+        if not isinstance(self.language_code, basestring):
+            raise utils.ValidationError(
+                'Expected language code to be a string, received %s' %
+                self.language_code)
+        if not any([self.language_code == lc['code']
+                    for lc in constants.ALL_LANGUAGE_CODES]):
+            raise utils.ValidationError(
+                'Invalid language code: %s' % self.language_code)
+
+        if not isinstance(self.skill_contents, SkillContents):
+            raise utils.ValidationError(
+                'Expected skill_contents to be a SkillContents object, ' +
+                'received %s' % self.skill_contents)
+        if not isinstance(self.misconceptions, list):
+            raise utils.ValidationError(
+                'Expected misconceptions to be a list, ' +
+                'received %s' % self.skill_contents)
+
+        self.skill_contents.validate()
```

Comment:
Hm, ordering here is a bit weird. Suggest: lines 373-376, then this line, then blank line, then lines 377-380, then lines 383-388. (I.e., fully validate each field one-by-one.)